src/Chainweb/Pact/PactService.hs

```diff
@@ -649,7 +650,15 @@ execLocal cwtx preflight sigVerify rdepth = withDiscardedBatch $ do
           -- withCheckpointerRewind as withCurrentCheckpointer
           -- (i.e. setting rewind to 0).
           | otherwise = Just 0
-        rewindHeader = Just $ _tcParentHeader ctx
+
+    let parentBlockHeader = _parentHeader $ _tcParentHeader ctx
+    let ancestorRank = fromIntegral $ _height $ _blockHeight parentBlockHeader - fromMaybe 0 rdepth
+    ancestor <- liftIO $ seekAncestor _psBlockHeaderDb parentBlockHeader ancestorRank
+
+    let rewindHeader
+          | Just a <- ancestor = Just $ ParentHeader a
+          -- when there is no ancestor, use the current parent
+          | otherwise = Just $ _tcParentHeader ctx
```

Comment:
I'm not sure about renaming the "rewind depth" to the "confirmation depth". Because the semantics is that we want to rewind the block, not to confirm something. At least that's what I have from user's POV.

Reply:
This makes sense to me. Rewind is the action, confirmation is an interpretation.

Reply:
+1 to both of u

Reply:
I think, on the external API "confirmation" is what the user wants. Internally, it's rewind.

So, user facing should be "confirmation depth" (it's a common term in Bitcoin). Rewind is an implementation details of the our checkpointer which has no meaning to external users.

Reply:
I've got some doubts about the "rewind depth" because it clashes with the checkpointer's rewind mechanism. We will get two different rewind semantics — the local command's rewind of blocks and checkpointer's rewind (the first uses the latter's).

We could say that the checkpointer stuff is an internal thing and users shouldn't know about it, but it's already leaked as an error message, see `PactException(RewindLimitExceeded)`. Also there is `reorgLimit` in `execValidateBlock`, which is a limit for checkpointer's rewind but with additional semantic.

Do we want our users to know about the checkpointer and how much? If not, then there will be two rewinds — 
"rewind depth" as an external API thing and "checkpointer rewinds" as an internal meaning, which we can try to hide. If yes, then we should be more clear and accurate with the terms to avoid ambiguity (still not a fan of "confirmation depth" in this context).

Reply:
I think, `confirmation depth` is a notion only at service REST API level.  "rewind depth" is a concept at checkpoint API level.

The idea is that in the future all service API endpoints that retrieve information from the blockchain would have a common "confirmation depth" parameter. This would include latest cut, querying headers, querying payloads, local calls, and possibly more.

"Rewind" wouldn't be a meaningful notion at the service API level, because it is relative to the checkpoint state which is hidden from the service API user. It also would not be meaningful with respect to querying cuts, headers, or payloads. Instead what service API users care about is the "level of confirmation" of the retrieved data, i.e. the depth in the chain with respect to the current (local) state of consensus.

Reply:
I wonder if we need to define more precisely what the notion of "current block" means within the context of the checkpointer. Is it just the current state of the checkpoint, or is the the latest state of consensus (which is external to the checkpointer). Clearly both are related, but the respective invariants are currently not very well documented or reflected in the code.